cmd/prometheus-postgresql-adapter/main.go

```diff
@@ -193,42 +195,41 @@ func buildClients(cfg *config) (writer, reader) {
 }
 
 func initElector(cfg *config, db *sql.DB) *util.Elector {
-	if cfg.restElection && cfg.haGroupLockId != 0 {
+	if cfg.restElection && cfg.haGroupLockID != 0 {
 		log.Error("msg", "Use either REST or PgAdvisoryLock for the leader election")
 		os.Exit(1)
 	}
 	if cfg.restElection {
 		return util.NewElector(util.NewRestElection())
 	}
-	if cfg.haGroupLockId != 0 {
-		if cfg.prometheusTimeout == -1 {
-			log.Error("msg", "Prometheus timeout configuration must be set when using PG advisory lock")
-			os.Exit(1)
-		}
-		lock, err := util.NewPgAdvisoryLock(cfg.haGroupLockId, db)
-		if err != nil {
-			log.Error("msg", "Error creating advisory lock", "haGroupLockId", cfg.haGroupLockId, "err", err)
-			os.Exit(1)
-		}
-		scheduledElector := util.NewScheduledElector(lock)
-		log.Info("msg", "Initialized leader election based on PostgreSQL advisory lock")
-		if cfg.prometheusTimeout != 0 {
-			go func() {
-				ticker := time.NewTicker(promLivenessCheck)
-				for {
-					select {
-					case <-ticker.C:
-						lastReq := atomic.LoadInt64(&lastRequestUnixNano)
-						scheduledElector.PrometheusLivenessCheck(lastReq, cfg.prometheusTimeout)
-					}
-				}
-			}()
-		}
-		return &scheduledElector.Elector
-	} else {
+	if cfg.haGroupLockID == 0 {
 		log.Warn("msg", "No adapter leader election. Group lock id is not set. Possible duplicate write load if running adapter in high-availability mode")
 		return nil
 	}
+	if cfg.prometheusTimeout == -1 {
+		log.Error("msg", "Prometheus timeout configuration must be set when using PG advisory lock")
+		os.Exit(1)
```

Comment:
perhaps in a follow up PR: maybe change this function to return an error, and then have the caller handle the error with `os.Exit(1)`